starport/services/scaffolder/module.go

```diff
@@ -213,36 +267,40 @@ func moduleExists(appPath string, moduleName string) (bool, error) {
 		return false, nil
 	}
 
-	return true, err
+	return err == nil, err
 }
 
-func checkModuleName(moduleName string) error {
+// checkModuleName checks if the name can be used as a module name
+func checkModuleName(appPath, moduleName string) error {
 	// go keyword
 	if token.Lookup(moduleName).IsKeyword() {
 		return fmt.Errorf("%s is a Go keyword", moduleName)
 	}
 
-	// name of default registered module
-	switch moduleName {
-	case
-		"auth",
-		"genutil",
-		"bank",
-		"capability",
-		"staking",
-		"mint",
-		"distr",
-		"gov",
-		"params",
-		"crisis",
-		"slashing",
-		"ibc",
-		"upgrade",
-		"evidence",
-		"transfer",
-		"vesting":
-		return fmt.Errorf("%s is a default module", moduleName)
+	// check if the name is a reserved name
+	if _, ok := reservedNames[moduleName]; ok {
```

Comment:
We should prevent the camel case names as well. like `Auth` and `Bank`.
Maybe we can convert the module name to lowercase before compare.

Reply:
Module names are only lowercase

Reply:
ohh, sorry, I saw [here](https://github.com/tendermint/starport/blob/6af942569ea6b63c0e2b3c22266675692d9c23be/starport/services/scaffolder/module.go#L139-L143), we convert it to lowercase before using